teuthology/salt.py

```diff
@@ -25,18 +23,21 @@ def __init__(self, ctx, config, **kwargs):
         # FIXME: this seems fragile (ens3 hardcoded)
         teuthology_ip_address = ifaddresses('ens3')[2][0]['addr']
         ip_addr = teuthology_ip_address.split('.')
-        teuthology_remote_name = "ubuntu@target{:03d}{:03d}{:03d}{:03d}.teuthology".format(
+        teuthology_remote_name = ('ubuntu@target{:03d}{:03d}{:03d}{:03d}'
+                                  '.teuthology').format(
             int(ip_addr[0]),
             int(ip_addr[1]),
             int(ip_addr[2]),
             int(ip_addr[3]),
         )
-        if(not config):
+        # If config has not master_remote attribute, salt is used by
+        # teuhtology for worker deployment and the Teuthology VM is master
```

Comment:
If config has no master_remote attribute, Salt is being used for worker deployment and the teuthology machine is the master.